pkg/apis/operator/v1alpha1/glustercluster_types.go

```diff
@@ -4,19 +4,62 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// GlusterClusterCredentials defines secret credentials
+type GlusterClusterCredentials struct {
+	SecretName      string `json:"secretName"`
+	SecretNamespace string `json:"secretNamespace"`
+}
+
+// GlusterStorageTarget defines a storage target
+type GlusterStorageTarget struct {
+	Name    string                    `json:"name"`
+	Address []string                  `json:"address"`
+	Creds   GlusterClusterCredentials `json:"credentials"`
+}
+
+// GlusterClusterReplicationDetails defines replication details
+type GlusterClusterReplicationDetails struct {
+	Creds   GlusterClusterCredentials `json:"credentials"`
+	Targets []GlusterStorageTarget    `json:"targets"`
+}
+
+// GlusterNodeThreshold defines threshold details for node
+type GlusterNodeThreshold struct {
+	Nodes          int    `json:"nodes"`
+	MinNodes       int    `json:"minNodes"`
+	MaxNodes       int    `json:"maxNodes"`
+	FreeStorageMin string `json:"freeStorageMin"`
+	FreeStorageMax string `json:"freeStorageMax"`
+}
+
+// GlusterNodeStorageDetails defines storage class details
+type GlusterNodeStorageDetails struct {
+	StorageClassName string `json:"storageClassName"`
+	Capacity         string `json:"capacity"`
+}
+
+// GlusterNodeTemplate defines a gluster node's template
+type GlusterNodeTemplate struct {
+	Name       string                    `json:"name"`
+	Zone       string                    `json:"zone"`
+	Thresholds GlusterNodeThreshold      `json:"thresholds"`
+	Affinity   string                    `json:"nodeAffinity"`
```

Comment:
I think this type should be imported from kube. It should be this structure: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#affinity-v1-core

Reply:
Ack